heartbeat/scheduler/scheduler.go

```diff
@@ -259,7 +266,6 @@ func (s *Scheduler) runRecursiveTask(jobCtx context.Context, task TaskFunc, wg *
 		return startedAt
 	default:
 		s.stats.activeTasks.Inc()
-		s.stats.waitingTasks.Dec()
```

Comment:
This caused a 'leak' in the counter since this case doesn't always run. Moved it up to right after the resource was acquired.